src/api/app/controllers/webui/users/notifications_controller.rb

```diff
@@ -0,0 +1,39 @@
+class Webui::Users::NotificationsController < Webui::WebuiController
+  before_action :require_login
+
+  def index
+    notification_type = params[:type]
+    case notification_type
+    when 'done'
+      @notifications = Notification.where(delivered: true)
+                                   .for_current_user(User.session)
+    when 'reviews'
+      @notifications = Notification.not_marked_as_done
+                                   .where(event_type: 'Event::ReviewWanted')
+                                   .for_current_user(User.session)
+    when 'comments'
+      @notifications = Notification.not_marked_as_done
+                                   .where(notifiable_type: 'Comment')
+                                   .for_current_user(User.session)
+    when 'state_changes'
+      @notifications = Notification.not_marked_as_done.where(notifiable_type: 'BsRequest')
+                                   .where.not(bs_request_oldstate: nil)
+                                   .for_current_user(User.session)
+    else
+      @notifications = Notification.not_marked_as_done
+                                   .for_current_user(User.session)
+    end
+  end
+
+  def update
+    notification = Notification.find(params[:id])
+    authorize notification, policy_class: NotificationPolicy
```

Comment:
nitpick: the class is inferred from the object...

Reply:
Currently a notification is still an object of the class "Notification::RssFeedItem". This will change in the future, but for now we have to call the policy explicitly